### PR TITLE
MULE-15064: WSConsumer not adding Content-Disposition Header when MTOM is enabled

### PR DESCRIPTION
**Pull request by mbuchwald**

### CODE REVIEW

File: modules/ws/src/main/java/org/mule/module/ws/consumer/WSConsumer.java [hunk 1]
```diff
@@ -474,7 +474,11 @@ private void copyAttachmentsRequest(MuleEvent event)
 
             for (String outboundAttachmentName : message.getOutboundAttachmentNames())
             {
-                Attachment attachment = new AttachmentImpl(outboundAttachmentName, message.getOutboundAttachment(outboundAttachmentName));
+                AttachmentImpl attachment = new AttachmentImpl(outboundAttachmentName, message.getOutboundAttachment(outboundAttachmentName));
+                if (mtomEnabled)
+                {
+                    attachment.setHeader("Content-Disposition", "attachment; name=\"" + outboundAttachmentName + "\"");
```
Comment (elrodro83) [inline modules/ws/src/main/java/org/mule/module/ws/consumer/WSConsumer.java]: use a constant from the http service

File: modules/ws/src/test/java/org/mule/module/ws/functional/MtomFunctionalTestCase.java [hunk 2]
```diff
@@ -56,6 +62,11 @@ public void uploadAttachmentTest() throws Exception
                           "<result>OK</result></ns2:uploadAttachmentResponse>";
 
         assertXMLEqual(expected, event.getMessage().getPayloadAsString());
+        assertNotNull(event.getMessage().getInvocationProperty("cxf_attachments"));
```
Comment (elrodro83) [inline modules/ws/src/test/java/org/mule/module/ws/functional/MtomFunctionalTestCase.java]: hamcrest

File: modules/ws/src/test/java/org/mule/module/ws/functional/MtomFunctionalTestCase.java [hunk 3]
```diff
@@ -56,6 +62,11 @@ public void uploadAttachmentTest() throws Exception
...
+        AttachmentImpl collection[] = ((Collection<AttachmentImpl>) event.getMessage().getInvocationProperty("cxf_attachments"))
+                .toArray(new AttachmentImpl[] {});
+        assertThat(collection.length, is(1));
```
Comment (elrodro83) [inline modules/ws/src/test/java/org/mule/module/ws/functional/MtomFunctionalTestCase.java]: use IsCollectionWithSize matcher

File: modules/ws/src/test/java/org/mule/module/ws/functional/MtomFunctionalTestCase.java [hunk 4]
```diff
@@ -56,6 +62,11 @@ public void uploadAttachmentTest() throws Exception
...
+        assertThat(collection[0].getHeader("Content-Disposition"), containsString("attachment"));
```
Comment (elrodro83) [inline modules/ws/src/test/java/org/mule/module/ws/functional/MtomFunctionalTestCase.java]: use a collection matcher

File: modules/ws/src/test/java/org/mule/module/ws/functional/MtomFunctionalTestCase.java [hunk 5]
```diff
@@ -56,6 +62,11 @@ public void uploadAttachmentTest() throws Exception
                           "<result>OK</result></ns2:uploadAttachmentResponse>";
 
         assertXMLEqual(expected, event.getMessage().getPayloadAsString());
+        assertNotNull(event.getMessage().getInvocationProperty("cxf_attachments"));
+        AttachmentImpl collection[] = ((Collection<AttachmentImpl>) event.getMessage().getInvocationProperty("cxf_attachments"))
```
Comment (elrodro83) [inline modules/ws/src/test/java/org/mule/module/ws/functional/MtomFunctionalTestCase.java]: why do you need an array and not jsut a collection?

File: modules/ws/src/test/java/org/mule/module/ws/functional/MtomFunctionalTestCase.java [hunk 6]
```diff
@@ -86,6 +97,11 @@ public void echoAttachment() throws Exception
         event = flow.process(event);
 
         assertAttachmentInResponse(event.getMessage(), TEST_FILE_ATTACHMENT);
+        assertNotNull(event.getMessage().getInvocationProperty("cxf_attachments"));
```
Comment (elrodro83) [inline modules/ws/src/test/java/org/mule/module/ws/functional/MtomFunctionalTestCase.java]: same as above

Comment (mbuchwald) [reply modules/ws/src/test/java/org/mule/module/ws/functional/MtomFunctionalTestCase.java]: also, extract to another function

File: modules/ws/src/test/java/org/mule/module/ws/functional/MtomFunctionalTestCase.java [hunk 7]
```diff
@@ -123,5 +132,36 @@ private String extractAttachmentId(String payload)
...
+            return attachment.getHeader(CONTENT_DISPOSITION).contains("attachment");
+        }
+
+        @Override
+        public void describeMismatch(Object o, Description description) {
+            description.appendText("Attachment doesn't have Content-Disposition Header, or it's not defined with the value 'attachment'");
+        }
+
+        @Override
+        public void _dont_implement_Matcher___instead_extend_BaseMatcher_() {
```
Comment (elrodro83) [inline modules/ws/src/test/java/org/mule/module/ws/functional/MtomFunctionalTestCase.java]: what this line says :)

Comment (mbuchwald) [reply modules/ws/src/test/java/org/mule/module/ws/functional/MtomFunctionalTestCase.java]: https://www.youtube.com/watch?v=Ya9sjHUoBs8

**Pull request merged**